src/Aspire.Hosting.AWS.DynamoDBLocal/DynamoDBLocalResourceBuilderExtensions.cs

```diff
@@ -0,0 +1,163 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net.Sockets;
+using System.Text.Json;
+using Amazon.DynamoDBv2;
+using Amazon.Runtime;
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.AWS.DynamoDBLocal;
+
+namespace Aspire.Hosting;
+
+public static class DynamoDBLocalResourceBuilderExtensions
+{
+    /// <summary>
+    /// Add an instance of DynamoDB local. This is a container pulled from Amazon ECR public gallery.
+    /// Projects that use DynamoDB local can get a reference to the instance using the WithAWSDynamoDBLocalReference method.
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <param name="name">The name of the resource.</param>
+    /// <param name="image">Optional: default value is "public.ecr.aws/aws-dynamodb-local/aws-dynamodb-local"</param>
+    /// <param name="tag">Optional: default value is "latest"</param>
+    /// <param name="disableDynamoDBLocalTelemetry">Optional: default to false. It controls setting the DynamoDB local DDB_LOCAL_TELEMETRY environment variable.</param>
+    /// <returns></returns>
+    /// <exception cref="DistributedApplicationException"></exception>
+    public static IResourceBuilder<IDynamoDBLocalResource> AddAWSDynamoDBLocal(this IDistributedApplicationBuilder builder,
+        string name, string image = "public.ecr.aws/aws-dynamodb-local/aws-dynamodb-local", string tag = "latest", bool disableDynamoDBLocalTelemetry = false)
```

Comment:
Realistically how likely is it that folks will want to change this image? Originally I used to want to put all kinds of options on these extension methods but I realized that less is more.

If someone really wants to change this they can replace the image annotation.

Reply:
I would expect a very 9X% percent use case. But if a user is using this in an air gapped region or private VPC then it is reasonable to override these values.

Reply:
@davidfowl  This does raise the question I have had. 

For hosting packages is it going to be considered a breaking change if we add new optional parameters to these methods in the future? In our SDK we consider that a binary breaking change and we avoid that by having methods always take in an options class instead of using parameters so we can always add new properties to the options class. So I get nervous relying on optional parameters but I see that is the convention being used. Is it not a breaking concern because the hosting packages are so top level that any changes would just get picked up in the recompile?

Reply:
There are few docs that may help answering (some of) your questions:
* https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/breaking-change-definitions.md
* https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/breaking-change-rules.md